pwiz_tools/Skyline/Model/SrmDocument.cs

```diff
@@ -2132,6 +2130,30 @@ public double GetCollisionEnergy(PeptideDocNode nodePep, TransitionGroupDocNode
             return ce ?? 0.0;
         }
 
+        private static double? GetExplicitCollisionEnergy(TransitionGroupDocNode nodeGroup, TransitionDocNode nodeTran)

```

Comment:
Can you just use returns instead of the ce = null at the top. Then just return null at the bottom?

Reply:
Just try to make it as simple as possible now. This is a mess function, when you allow for simply returning as soon as you have a value.

Reply:
perhaps you'll prefer this next formulation.

Reply:
I guess it wasn't as easy to simplify as I had been thinking. I took a closer look in VS and did my best to get rid of the cruft, which did end up using a double? ce = null at the start of the function and a single return. Good to go now.

Reply:
By the way, I got rid of all the (TransitionDocNode) casting by using TransitionGroup.Transitions instead of TransitionGroup.Children, but even with the latter, you could have used Children.Cast<TransitionDocNode>() for cleaner code.